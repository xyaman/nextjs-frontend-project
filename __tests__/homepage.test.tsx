import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import HomeLayout from "@/app/layout";

import EditPage from "@/app/edit/[id]/page";
import AddPage from "@/app/edit/page";

// Mock useRouter:
jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      prefetch: () => null,
    };
  },
  usePathname: () => "/",
}));

// Navbar should be server-side rendered
describe("Page", () => {
  it("navbar text", () => {
    render(<HomeLayout children={<></>} />);
    expect(screen.getByText("NextJs Frontend")).toBeInTheDocument();
  });
});

// The default behaiour of the test is to return load screen
// because error should be null and users.length should be 0, so it
// should return loading screen
describe("Edit users", () => {
  it("should has one progressbar (circular progress)", () => {
    render(<EditPage params={{ id: "1" }} />);
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });
});

// New users page also should fetch all users firsts
describe("New", () => {
  it("should has one progressbar (circular progress)", () => {
    render(<AddPage />);
    expect(screen.getByRole("progressbar")).toBeInTheDocument();
  });
});
