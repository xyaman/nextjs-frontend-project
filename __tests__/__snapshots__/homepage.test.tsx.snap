// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1t6e35q-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <a
          href="/"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeLarge css-19f4tkl-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          NextJs Frontend
        </div>
      </div>
    </header>
  </div>
  <footer
    class="MuiBox-root css-22baet"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
    >
      Made by Diego Barria
    </h6>
    <p
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter css-1xatcrf-MuiTypography-root"
    >
      INYOU Test Project - 2024
    </p>
  </footer>
  <button
    aria-label="add"
    class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default css-18oex0m-MuiButtonBase-root-MuiFab-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AddIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Snapshots should match snapshot 2`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-13e1wmu-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-bx44ri"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ee3l2n-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshots should match snapshot 3`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-13e1wmu-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-bx44ri"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1ee3l2n-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
